ArchUnitNET/Fluent/Slices/SlicesShould.cs

```diff
@@ -0,0 +1,125 @@
+﻿//  Copyright 2019 Florian Gather <florian.gather@tngtech.com>
+// 	Copyright 2019 Paula Ruiz <paularuiz22@gmail.com>
+// 	Copyright 2019 Fritz Brandhuber <fritz.brandhuber@tngtech.com>
+// 
+// 	SPDX-License-Identifier: Apache-2.0
+// 
+
+using System.Collections.Generic;
+using System.Linq;
+using ArchUnitNET.Domain;
+using StronglyConnectedComponents;
+
+namespace ArchUnitNET.Fluent.Slices
+{
+    public class SlicesShould
+    {
+        private readonly SliceRuleCreator _ruleCreator;
+
+        public SlicesShould(SliceRuleCreator ruleCreator)
+        {
+            _ruleCreator = ruleCreator;
+        }
+
+        public SliceRule BeFreeOfCycles()
+        {
+            IEnumerable<EvaluationResult> Evaluate(IEnumerable<Slice> slices, ICanBeEvaluated archRule,
+                Architecture architecture)
+            {
+                var slicesList = slices.ToList();
+
+                IEnumerable<Slice> FindDependencies(Slice slice)
```

Comment:
we have a local function in a local function in a method.. and we need to go deeper :-)

Reply:
changed to class function